File: minseong/hash/전화번호목록.js
function solution(phone_book) {
  // 전화번호를 사전 순으로 정렬
  phone_book.sort();

  // 정렬된 목록에서 연속된 두 번호를 검사
  for (let i = 0; i < phone_book.length - 1; i++) {
    if (phone_book[i + 1].startsWith(phone_book[i])) {
      return false;
    }
  }

  return true;
}

File: Grace/hash/전화번호 목록.py
def solution(phone_book):
    # 1. Hash map을 만든다
    hash_map = {}
    for phone_number in phone_book:
        hash_map[phone_number] = 1
    
    for phone_number in phone_book:
        jubdoo = ""
        for number in phone_number:
            jubdoo += number
            
            if jubdoo in hash_map and jubdoo != phone_number:
                return False
    return True